frontend-new/src/auth/components/PasswordRequirements/PasswordRequirements.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { Box, Typography } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+
+interface PasswordRequirementsProps {
+  validationResults: {
+    length: boolean;
+    lowercase: boolean;
+    uppercase: boolean;
+    number: boolean;
+    specialChar: boolean;
+  };
+}
+
+const PasswordRequirements: React.FC<PasswordRequirementsProps> = ({ validationResults }) => {
+  const theme = useTheme();
+
+  const validationCriteria = [
+    { label: "Password must be at least 8 characters long.", isValid: validationResults.length },
```

Comment:
perhaps name this passwordLength or something similar? see how `validationResults.length` can be a bit confusing? whether you mean the length field or an array.length kinda thing... maybe?

Reply:
or making the booleans start with 'is/has' is also an option... like hasLowercase, hasNumber, isLongEnough etcetera

Reply:
just a thought